lib/Dancer2/Manual/Keywords.pod

```diff
@@ -1146,6 +1180,108 @@ URL encoding via a third parameter:
     uri_for('/path', { foo => 'qux%3Dquo' }, 1);
     # would return http://localhost:5000/path?foo=qux%3Dquo
 
+=head2 uri_for_route
+
+An enhanced version of C<uri_for> that utilizes their names.
+
+    get 'view_entry' => '/entry/view/:id' => sub {...};
+
+Now that the route has a name we can use C<uri_for_route> to
+create a URI for it:
+
+    my $path = uri_for_route(
+        'view_entry',
+        { 'id'  => 3 },
+        { 'foo' => 'bar' },
+    );
+
+    # (assuming it's run on a local server in HTTP port 5000)
+    # $path = 'http://localhost:5000/entry/view/3?foo=bar'
+
+This works for every HTTP method, except C<HEAD> (which is
+effectively a C<GET>).
+
+There are multiple arguments options:
+
+=over 4
+
+=item * Route parameters
+
+The first argument controls the route parameters:
+
+    get 'test' => '/:foo/:bar' => sub {1};
+    # ...
+    $path = uri_for_route( 'test', { 'foo' => 'hello', 'bar' => 'world' } );
+    # $path = http://localhost:5000/hello/world
+
+=item * Splat route parameters
+
+If you provide an arrayref instead of hashref, it will assume on
+these being splat and megasplat args:
+
+    get 'test' => '/*/*/**' => sub {1};
+    # ...
+    $path = uri_for_route(
+        'test',
+        [ 'hello', 'world', [ 'myhello', 'myworld' ],
+    );
+    # $path = http://localhost:5000/hello/world/myhello/myworld
+
+=item * Mixed route parameters
+
+If you have a route that includes both, the plat and megasplat
+arguments need to be under the C<splat> key:
+
+    patch 'test' => '/*/:id/*/:foo/*' => sub {1};
+    # ...
+    $path = uri_for_route(
+        'test',
+        {
+            'id'    => 4,
+            'foo  ' => 'bar',
+            'splat' => [ 'hello', 'world' ],
+        }
+    );
+    # $path = http://localhost:5000/hello/4/world/bar
+
+=item * Query parameters
+
+If you want to create a path the query parameters, use the
+second argument:
+
+    get 'index'       => '/:foo' => sub {1};
+    get 'update_form' => '/update' => sub {1};
+
+    # ...
+
+    $path = uri_for_route(
+        'index',
+        { 'foo' => 'bar' },
+        { 'id'  => 1 },
+    );
+    # $path = http://localhost:5000/bar?id=1
+
+    $path = uri_for_route( 'update_form', {}, { 'id' => 2 } );
+    # $path = http://localhost:5000/update?id=2
+
+(Technically, only C<GET> requests should include query parameters, but
+C<uri_for_route> does not enforce this.)
+
+=item * Escaping
+
+The final parameter determines whether the URI will be URI-escaped:
+
+    get 'show_entry' => '/view/:str_id' => sub {1};
+    # ...
+    $path = uri_for_route( 'show_entry' => { 'str_id' => '!£%^@' }, {}, 1 );
```

Comment:
The `1` is the `$dont_escape` parameter, but this is showing the result as URI-escaped.  I think the documentation of this parameter needs to be flipped around --- default == escape, `1` == don't escape.  But I could be missing something!

Reply:
Yeah, this confused me a bit too. I decided to keep this with `uri_for()` so it's alas, a disabling flag that - when set to true - will disable.

I fixed the docs though!